Geometry/TrackerCommonData/plugins/dd4hep/DDTrackerRingAlgo.cc

```diff
@@ -0,0 +1,129 @@
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/GeantUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+using namespace std;
+using namespace dd4hep;
+using namespace cms;
+using namespace geant_units::operators;
+
+static long algorithm(Detector& /* description */,
```

Comment:
It seems like this should go in an anonymous namespace since the function name is very generic. Maybe I misunderstand something about the design pattern of dd4hep.